### ISSUE TITLE
Trouble with pinned certificates

### ISSUE THREAD

**Issue by pfiggins** [justcoding121/Titanium-Web-Proxy#108]
I'm having some trouble with pinned certificates. So far I've noticed it with dropbox and google hangouts desktop application. As shown in the example, I tried excluding dropbox like so -
`ExcludedHttpsHostNameRegex = new List<string>() { "dropbox.com" }`

I've also tried a wildcard approach. I don't know regex too well, but figured I'd give it a shot. -
`ExcludedHttpsHostNameRegex = new List<string>() { "((.*)dropbox.com)" }`

Then I thought to step through the code and noted that the code below does work and the result is false -

```
var excluded = endPoint.ExcludedHttpsHostNameRegex != null ?
                    endPoint.ExcludedHttpsHostNameRegex.Any(x => Regex.IsMatch(httpRemoteUri.Host, x)) : false;
```

BUT... If I pause and then restart dropbox sync from the dropbox icon, it fails to make an SSL connection.

If I exit my app, dropbox begins working again. 

Any idea why my calls to dropbox.com aren't passing through?

**Comment by justcoding121** [comment]
Interesting. I will look at it asap.

**Comment by pfiggins** [comment]
Scratch that. I had changed the cert store to Local Machine. I'm re-using a root certificate that our app already uses for other things, and it happens to be in the LM store. That seemed to be the root cause of the issue. 

On a related note, I also found that setting both stores to Local Machine actually causes a deadlock on the call to makecert. I can step through and see it calling tcs.SetResult, but the await never continues in ProxyServer.cs. Switching MyStore back to Current User fixed both issues, but for the life of me, I don't see why it's deadlocking there with MyStore set to Local Machine.
